### PR TITLE
Sharing more code for light data building (RT) and building a more optimal atlas

### PR DESCRIPTION
**Pull request by anisunity**
- Sharing GetLightData code between rasterization and ray tracing to avoid duplication and lack of features
- Inject all lights at the same time (rasterization and ray tracing) into the cookie atlas to avoid waste of space.
- Added a test scene to make sure it works properly

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 1]
```diff
@@ -1148,7 +1148,7 @@ static Vector3 EvaluateAtmosphericAttenuation(PhysicallyBasedSky skySettings, Ve
 
             lightData.shadowDimmer           = additionalLightData.shadowDimmer;
             lightData.volumetricShadowDimmer = additionalLightData.volumetricShadowDimmer;
-            GetContactShadowMask(additionalLightData, HDAdditionalLightData.ScalableSettings.UseContactShadow(m_Asset), hdCamera, ref lightData.contactShadowMask,ref lightData.isRayTracedContactShadow);
+            GetContactShadowMask(additionalLightData, HDAdditionalLightData.ScalableSettings.UseContactShadow(m_Asset), hdCamera, true, ref lightData.contactShadowMask,ref lightData.isRayTracedContactShadow);
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: This flying booleans I find hard to read, can you use the  C# syntax to do 

isRasterization: true ?

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 2]
```diff
@@ -1476,13 +1474,18 @@ bool EnoughScreenSpaceShadowSlots(GPULightType gpuLightType, int screenSpaceChan
...
+            if (isRasterization)
+            {
+                // Keep track of the shadow map (for indirect lighting and transparents)
+                additionalLightData.shadowIndex = shadowIndex;
+            }
+
 
             //Value of max smoothness is derived from Radius. Formula results from eyeballing. Radius of 0 results in 1 and radius of 2.5 results in 0.
-            float maxSmoothness = Mathf.Clamp01(1.1725f / (1.01f + Mathf.Pow(1.0f * (additionalLightData.shapeRadius + 0.1f), 2f)) - 0.15f);
+            //float maxSmoothness = Mathf.Clamp01(1.1725f / (1.01f + Mathf.Pow(1.0f * (additionalLightData.shapeRadius + 0.1f), 2f)) - 0.15f);
```
Comment (FrancescoC-unity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: Why is it commented out?

Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: Mistake while debugging, good catch thanks

Comment (anisunity) [reply com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs]: fixed

**Comment by iM0ve** [comment]
Currently, the new test is failing due to missing reference images. All other tests pass
Capture

File: com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs [hunk 3]
```diff
@@ -1285,17 +1285,13 @@ bool EnoughScreenSpaceShadowSlots(GPULightType gpuLightType, int screenSpaceChan
         }
 
         internal void GetLightData(CommandBuffer cmd, HDCamera hdCamera, HDShadowSettings shadowSettings, VisibleLight light, Light lightComponent,
-            int lightIndex, int shadowIndex, BoolScalableSetting contactShadowsScalableSetting, ref Vector3 lightDimensions, ref int screenSpaceShadowIndex, ref int screenSpaceChannelSlot)
+            ProcessedLightData processedData, int shadowIndex,  BoolScalableSetting contactShadowsScalableSetting, bool isRasterization, ref Vector3 lightDimensions, ref int screenSpaceShadowIndex, ref int screenSpaceChannelSlot, ref LightData lightData)
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/Lighting/LightLoop/LightLoop.cs:1288]: Might want to add the "in" qualifier to processedData parameter as it's quite a big struct iirc.

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 4]
```diff
@@ -1886,10 +1886,11 @@ AOVRequestData aovRequest
 
             if (m_RayTracingSupported)
             {
-                // This call need to happen once per camera
+                // This calls need to happen once per camera
```
Comment (JulienIgnace-Unity) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs:1889]: Mistype? "Call" was right here.

**Pull request merged**